src/middlewared/middlewared/plugins/enclosure.py

```diff
@@ -497,23 +489,21 @@ def _set_model(self, data):
         elif R_SERIES_REGEX.match(self.encname) or R20A_REGEX.match(self.encname):
             self.model = self.product_name.replace("TRUENAS-", "")
             self.controller = True
-            if self.model in ["R20", "R20A"]:
-                self.model = f"{self.model}, Drawer #1"
         elif (
-            self.product_name in ["TRUENAS-R20", "TRUENAS-R20A"] and
             self.encname == "AHCI SGPIO Enclosure 2.00" and
-            len(data.splitlines()) == 6
+            self.system_info["system_product"] in ["TRUENAS-R20", "TRUENAS-R20A"]
         ):
-            self.model = f"{self.product_name.replace('TRUENAS-', '')}, Drawer #2"
+            self.model = self.product_name.replace("TRUENAS-", "")
+            self.controller = True
+        elif self.encname.startswith("iX TrueNAS R20p"):
```

Comment:
Let's make this its own `R20_REGEX` and then move the check inline to line 489.

```
elif R_SERIES_REGEX.match(self.encname) or R20A_REGEX.match(self.encname) or R20_REGEX.match(self.encname):

Reply:
Should be resolved now